tests/fsstress/fsstress_test.go

```diff
@@ -0,0 +1,364 @@
+
+	var appInstanceConfig *config.AppInstanceConfig
+
+	for _, id := range edgeNode.GetApplicationInstances() {
+		appConfig, _ := tc.GetController().GetApplicationInstanceConfig(id)
+		if appConfig.Displayname == appName {
+			appInstanceConfig = appConfig
+			break
+		}
+	}
+	return appInstanceConfig
+}
+
+//TestAccess checks if SSH of app is accessible
+func TestAccess(t *testing.T) {
+
+	edgeNode := tc.GetEdgeNode(tc.WithTest(t))
+
+	setAppName()
+
+	appInstanceConfig := getAppInstanceConfig(edgeNode, appName)
+
+	if appInstanceConfig == nil {
+		t.Fatalf("No app found with name %s", appName)
+	}
+
+	appID, err := uuid.FromString(appInstanceConfig.Uuidandversion.Uuid)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log(utils.AddTimestamp("Add function to obtain EVE IP"))
+
+	tc.AddProcTimer(edgeNode, getEVEIP(edgeNode))
+
+	t.Log(utils.AddTimestamp("Add trying to access SSH of app"))
+
+	tc.AddProcTimer(edgeNode, projects.SendCommandSSH(&externalIP, sshPort, "ubuntu", "passw0rd", "exit", true, defaults.DefaultRepeatTimeout))
+
+	tc.ExpandOnSuccess(int(expand.Seconds()))
+
+	callback := func() {
+		fmt.Printf("--- app %s logs ---\n", appInstanceConfig.Displayname)
+		if err = tc.GetController().LogAppsChecker(edgeNode.GetID(), appID, nil, eapps.HandleFactory(eapps.LogJSON, false), eapps.LogExist, 0); err != nil {
+			t.Fatalf("LogAppsChecker: %s", err)
+		}
+		fmt.Println("------")
+	}
+
+	tc.WaitForProcWithErrorCallback(int(timewait.Seconds()), callback)
+
+}
+
+//TestRunStress run fsstress test on guest vm
+func TestRunStress(t *testing.T) {
+	edgeNode := tc.GetEdgeNode(tc.WithTest(t))
+	timeTestStart := time.Now()
+	setAppName()
+
+	appInstanceConfig := getAppInstanceConfig(edgeNode, appName)
+
+	if appInstanceConfig == nil {
+		t.Fatalf("No app found with name %s", appName)
+	}
+
+	appID, err := uuid.FromString(appInstanceConfig.Uuidandversion.Uuid)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pathScript := *scriptpath
+	if *scriptpath == "" {
+		pathScript = filepath.Join(viper.GetString("eden.tests"), "/fsstress/testdata/run-script.sh")
+	}
+
+	tc.AddProcTimer(edgeNode, getEVEIP(edgeNode))
+
+	t.Log(utils.AddTimestamp("Send script on guest VM"))
+	projects.SendFileSCP(&externalIP, sshPort, "ubuntu", "passw0rd", pathScript, "/home/ubuntu/run-script.sh")
+
+	t.Log(utils.AddTimestamp("Add options for running"))
+	projects.SendCommandSSH(&externalIP, sshPort, "ubuntu", "passw0rd", "chmod +x ~/run-script.sh", true, defaults.DefaultRepeatTimeout)
+
+	t.Log(utils.AddTimestamp("Run script on guest VM"))
+	projects.SendCommandSSH(&externalIP, sshPort, "ubuntu", "passw0rd", "~/run-script.sh &", false, defaults.DefaultRepeatTimeout)
+
+	tc.AddProcTimer(edgeNode, projects.SendCommandSSH(&externalIP, sshPort, "ubuntu", "passw0rd", "exit", true, defaults.DefaultRepeatTimeout))
+
+	tc.ExpandOnSuccess(int(expand.Seconds()))
+
+	t.Log(utils.AddTimestamp("Waiting sucсess finished test"))
+	for {
```

Comment:
As I can see, we will sit in this loop until time is near to out. But seems, we want to check is our VM running or is it respond to ssh or is our stress script running.
I think it is reasonable to write projects.ProcTimerFunc with some check (say, ssh) and your time check inside and remove this loop.